build-tools/mono-runtimes/mono-runtimes.projitems

```diff
@@ -160,4 +206,226 @@
       <OutputMonoPosixHelperFilename>libMonoPosixHelper</OutputMonoPosixHelperFilename>
     </_MonoRuntime>
   </ItemGroup>
+
+  <ItemGroup>
+    <!-- LLVM
+
+         Note: due to an xbuild bug we can't put property references inside item metadata that is to be used in the Target.Outputs attribute.
+         In this case the affected metadata item is BuildDir. Once the bug is fixed, we should replace the hard-coded values in BuildDir
+         metadata items with references to the appropriate $(_LlvmBuildDir*) properties. The properties in question are defined in the
+         mono-runtimes.props file.
+         At this moment the BuildDir item is treated as a *subdirectory* name and once the bug is fixed it will contain a full directory path - that
+         means mono-rutime.targets which use BuildDir (_ConfigureLlvm, BuildLlvm and _InstallLlvm) must be updated accordingly.
+    -->
+    <_LlvmRuntime Include="llvm32" Condition=" '$(_LlvmNeeded)' != '' ">
+      <BuildDir>build-32</BuildDir>
+      <Prefix>$(_LlvmPrefix32)</Prefix>
+      <ConfigureFlags>$(_LlvmConfigureFlags32)</ConfigureFlags>
+      <ConfigureEnvironment></ConfigureEnvironment>
+      <BuildEnvironment></BuildEnvironment>
+      <ExeSuffix></ExeSuffix>
+      <InstallBinaries>true</InstallBinaries>
+    </_LlvmRuntime>
+
+    <_LlvmRuntime Include="llvm64" Condition=" '$(_LlvmNeeded)' != '' And '$(_LlvmCanBuild64)' == 'yes' ">
+      <BuildDir>build-64</BuildDir>
+      <Prefix>$(_LlvmPrefix64)</Prefix>
+      <ConfigureFlags>$(_LlvmConfigureFlags64)</ConfigureFlags>
+      <ConfigureEnvironment></ConfigureEnvironment>
+      <BuildEnvironment></BuildEnvironment>
+      <ExeSuffix></ExeSuffix>
+      <InstallBinaries>true</InstallBinaries>
+    </_LlvmRuntime>
+
+    <_LlvmRuntime Include="llvmwin32" Condition=" '$(_LlvmNeeded)' != '' And $(AndroidSupportedHostJitAbisForConditionalChecks.Contains (':mxe-Win32:')) ">
+      <BuildDir>build-win32</BuildDir>
```

Comment:
For each LLVM build we use two directories:

   * build-XYZ
   * installed-XYZ

and originally both of them were stored in properties defined in the .props file and the BuildDir property was used here to define the value of `BuildDir` - which is nicer than having the property used in `Include` which here serves as a descriptive name. However, due to an `xbuild` bug described in the comment above we now use the hard-coded value until `xbuild` is fixed or replaced by `msbuild`. Also, I don't like abusing `%Identity` - this metadata name is, rather, confusing as it doesn't tell the person working with the code what it really means. Using `%(_LlvmRuntime.BuildDir)` is nicely descriptive and clear.